### PR TITLE
feat: generalize tracking feature towards 'Frappe Tag Manager'

### PR DESCRIPTION
**Pull request by blaggacao**
see writeup: https://github.com/frappe/builder/issues/62

companion to https://github.com/frappe/frappe/pull/24466

lmkwyt

### CODE REVIEW

File: builder/templates/generators/webpage.html [hunk 1]
```diff
@@ -41,10 +41,13 @@
...
 	<!-- csrf_token -->
 	<script async>
-		if (navigator.doNotTrack != 1) {document.addEventListener("DOMContentLoaded", function() {let b=getBrowser(),q=getQueryParams();import('https://openfpcdn.io/fingerprintjs/v3').then(f=>f.load()).then(fp=>fp.get()).then(r=>{const d={referrer:document.referrer,browser:b.name,version:b.version,user_tz:Intl.DateTimeFormat().resolvedOptions().timeZone,source:q.source,medium:q.medium,campaign:q.campaign,visitor_id:r.visitorId};makeViewLog(d)})})}function getBrowser(){const ua=navigator.userAgent,b={};if(ua.indexOf("Chrome")!==-1){b.name="Chrome";b.version=parseInt(ua.split("Chrome/")[1])}else if(ua.indexOf("Firefox")!==-1){b.name="Firefox";b.version=parseInt(ua.split("Firefox/")[1])}else{b.name="Unknown";b.version="Unknown"}return b}function getQueryParams(){const q={},p=window.location.search.substring(1).split("&");p.forEach(p=>{const [k,v]=p.split("=");q[k]=v});return q}function makeViewLog(d){fetch('/api/method/frappe.website.doctype.web_page_view.web_page_view.make_view_log',{method:'POST',headers:{'Content-Type':'application/json',"X-Frappe-CSRF-Token": frappe.csrf_token},body:JSON.stringify(d)})}
+		frappe.fel("WebPageView")
```
Comment (blaggacao) [inline builder/templates/generators/webpage.html:50]: I didn't want to name it `frappe.track` as all tag managers that I have seen don't seem to be as explicit, either.

So here: "Frappe Event Log" or `fel`, in short.

**Comment by blaggacao** [comment]
Alternatively something like matomo does with an array that is the worked on might also be a good idea to have events immediately stacked and the pushed to the server later (after the script is loaded):

```js
var _mtm = window._mtm = window._mtm || [];
_mtm.push({'mtm.startTime': (new Date().getTime()), 'event': 'mtm.Start'});
(function() {
    var d=document, g=d.createElement('script'), s=d.getElementsByTagName('script')[0];
    g.async=true; g.src='https://cdn.matomo.cloud/<my-site>.matomo.cloud/container_<hash>.js'; s.parentNode.insertBefore(g,s);
})();
```

**Comment by blaggacao** [comment]
ping

**Comment by blaggacao** [comment]
> Go for it now. The future is promised to no one.   - Wayne Dyer

---

<sub>Kindly help move this PR forward. Many thanks from Yours Sincerely!</sub>

**Comment by blaggacao** [comment]
> What the caterpillar calls the end of the world, the master calls a butterfly.  - Richard Bach

---

<sub>Kindly help move this PR forward. Many thanks from Yours Sincerely!</sub>

**Comment by blaggacao** [comment]
> Watch your thoughts, they become your words Watch your words, they become your actions Watch your actions, they become your habits Watch your habits, they become your character Watch your character, it becomes your destiny.  - 

---

<sub>Kindly help move this PR forward. Many thanks from Yours Sincerely!</sub>

**Comment by blaggacao** [comment]
> You may say Im a dreamer, but Im not the only one, I hope someday you will join us, and the world will live as one.  - John Lennon

---

<sub>Kindly help move this PR forward. Many thanks from Yours Sincerely!</sub>

**Comment by blaggacao** [comment]
> I have just three things to teach: simplicity, patience, compassion. These three are your greatest treasures.   - Lao Tzu 

---

<sub>Kindly help move this PR forward. Many thanks from Yours Sincerely!</sub>

**Comment by blaggacao** [comment]
> The important thing is this: to be able at any moment to sacrifice what we are for what we could become.  - Charles Dubois

---

<sub>Kindly help move this PR forward. Many thanks from Yours Sincerely!</sub>

**Comment by blaggacao** [comment]
> Obstacles are those frightful things you see when you take your eyes off your goal.  - Henry Ford

---

<sub>Kindly help move this PR forward. Many thanks from Yours Sincerely!</sub>

**Comment by blaggacao** [comment]
> Our distrust is very expensive.  - Ralph Emerson

---

<sub>Kindly help move this PR forward. Many thanks from Yours Sincerely!</sub>

**Comment by blaggacao** [comment]
> An invincible determination can accomplish almost anything and in this lies the great distinction between great men and little men.  - Thomas Fuller 

---

<sub>Kindly help move this PR forward. Many thanks from Yours Sincerely!</sub>

**Comment by surajshetty3416** [comment]
Dependant PR is closed... also, unless the dependant feature is ported to V15, we can't merge this without extra handling.

Closing this for now.

**Comment by blaggacao** [comment]
@surajshetty3416 Thanks! Would you agree with the dependant feature? I'd be happy to revive; it's a hybrid between a full plausible integration and a lightwight built-in tracking. Though I've already explored how to connect both based on this data layout in frappe, so it would be a useful feature to explore.

Should I?

**Comment by blaggacao** [comment]
> Should I?

Would you support it? :smile:

**Comment by surajshetty3416** [comment]
> Would you support it? 😄

Sure, if everything is handled properly :)

**Pull request closed without merging**